shared-libs/rules-engine/src/provider-wireup.js

```diff
@@ -272,27 +289,28 @@ const refreshRulesEmissionForContacts = (provider, calculationTimestamp, contact
   });
 };
 
-const storeTargetsDoc = (provider, targets, filterInterval, force = false) => {
-  const targetDocTag = filterInterval ? moment(filterInterval.end).format('YYYY-MM') : 'latest';
+const storeTargetsDoc = (provider, aggregate, updatedTargets) => {
+  const targetDocTag = aggregate.filterInterval ? moment(aggregate.filterInterval.end).format('YYYY-MM') : 'latest';
   const minifyTarget = target => ({ id: target.id, value: target.value });
+  const userContext = {
+    userContactDoc: rulesStateStore.currentUserContact(),
+    userSettingsDoc: rulesStateStore.currentUserSettings(),
+  };
 
   return provider.commitTargetDoc(
-    targets.map(minifyTarget),
-    rulesStateStore.currentUserContact(),
-    rulesStateStore.currentUserSettings(),
+    aggregate.targets.map(minifyTarget),
     targetDocTag,
-    force
+    userContext,
+    updatedTargets
   );
 };
 
-// Because we only save the `target` document once per day (when we calculate targets for the first time),
-// we're losing all updates to targets that happened in the last day of the reporting period.
 // This function takes the last saved state (which may be stale) and generates the targets doc for the corresponding
 // reporting interval (that includes the date when the state was calculated).
 // We don't recalculate the state prior to this because we support targets that count events infinitely to emit `now`,
 // which means that they would all be excluded from the emission filter (being outside the past reporting interval).
 // https://github.com/medic/cht-core/issues/6209
-const handleIntervalTurnover = (provider, { monthStartDate }) => {
+const handleIntervalTurnover = async (provider, { monthStartDate }) => {
```

Comment:
i'm a bit unclear when `handleIntervalTurnover` is now necessary. since we write the target doc as soon as data changes

Reply:
It's not needed anymore, indeed. Gareth and I discussed it already, and we decided that due to the urgency on this, we would revisit it, and further cleanup of the library in a next iteration.

Reply:
There might be edge cases where this is still useful - ex: the user closes their app before saving the target doc and only opening the app in the old interval.

This was a talking point, about updating what `now` means and actually making it possible to completely recalculate targets within another interval, but the conclusion was that this change was not straightforward (decisions need to be made, like: do we keep a "now" that means "this emission should always be counted unless the date is before the thing that emits was created?" or is `now` a range?). 

I think we should revisit this once we can accurately recalculate the target counts for any interval.